### ISSUE TITLE
What triggers U2U1204

### ISSUE THREAD

**Issue by jnyrup** [u2uconsult/codeanalyzers#24]
What does it take to trigger U2U1204?
Below are two examples of what I would think as implicit conversions from `IQueryable` to `IEnumerable`, but none of them triggers U2U1204.

```C#
public void TryToTriggerU2U1204()
{
    IQueryable<string> queryable = Enumerable.Empty<string>().AsQueryable();

    // A method group cannot be converted to Expression<Func<T, bool>>
    // So it resolves to IEnumerable.Select instead of IQueryable.Select
    queryable.Select(Predicate);

    // implicit conversion to IEnumerable because of function signature
    ImplicitConvertToIEnumerable(queryable);
}

private bool Predicate(string str) => true;

private static void ImplicitConvertToIEnumerable<T>(IEnumerable<T> _) { }
```

**Comment by KrisVandermotten** [comment]
U2U1204 currently triggers on

1. assignment expressions where a  value of an `IQueryable<T>` type is assigned to a variable of an `IEnumerable<T>` type
2. variable declarations of an `IEnumerable<T>` type that are initialized with a value of an `IQueryable<T>` type
3. a function with a result of an `IEnumerable<T>` type returns a value of an `IQueryable<T>` type

An implicit conversion of an argument does not trigger U2U1204, because that would trigger too many false positives. For example, it would trigger on `queryable.ToList()`.

**Comment by jnyrup** [comment]
Thanks for the explanation 👍
